org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/validation/ValidatorFragment2.xtend

```diff
@@ -123,38 +127,48 @@ class ValidatorFragment2 extends AbstractInheritingFragment {
 			}
 		''').writeTo(projectConfig.runtime.src)
 	}
-	
+
 	protected def generateGenValidator() {
 		// take the non-abstract class signature for the src-gen class in case of !generateStub
-		//  as validators of sub languages refer to 'superGrammar.validatorClass',
-		//  see 'getGenValidatorSuperClass(...)'
-		val genClass = if (isGenerateStub) grammar.abstractValidatorClass else grammar.validatorClass
+		// as validators of sub languages refer to 'superGrammar.validatorClass',
+		// see 'getGenValidatorSuperClass(...)'
+		val genClass = if(isGenerateStub) grammar.abstractValidatorClass else grammar.validatorClass
 
 		val javaFile = fileAccessFactory.createGeneratedJavaFile(genClass)
-		
 		javaFile.content = '''
 			�IF !composedChecks.empty�
-			@�ComposedChecks�(validators = {�FOR validator: composedChecks SEPARATOR ", "��validator.typeRef�.class�ENDFOR�})
+				@�ComposedChecks�(validators = {�FOR validator: composedChecks SEPARATOR ", "��validator.typeRef�.class�ENDFOR�})
 			�ENDIF�
 			public �IF isGenerateStub�abstract �ENDIF�class �genClass.simpleName� extends �grammar.genValidatorSuperClass� {
 				
 				@Override
 				protected �List�<�EPackage�> getEPackages() {
 					�List�<�EPackage�> result = new �ArrayList�<�EPackage�>(�IF inheritImplementation && grammar.nonTerminalsSuperGrammar !== null�super.getEPackages()�ENDIF�);
-					�FOR e: generatedPackagesToValidate�
+					�FOR e : generatedPackagesToValidate�
 						result.add(�e.generatedEPackageName�.eINSTANCE);
 					�ENDFOR�
-					�FOR e: registryPackagesToValidate�
+					�FOR e : registryPackagesToValidate�
 						result.add(EPackage.Registry.INSTANCE.getEPackage("�e.nsURI�"));
-				 	�ENDFOR�
+					�ENDFOR�
 					return result;
 				}
-				
+				�generateValidationToDeprecateRules�
```

Comment:
i am not sure if this is what i thought of when considered deprecated.

Reply:
@kthoms @ArneDeutsch @szarnekow what do you think

Reply:
What do you mean?

Reply:
i am not sure if deprecated in grammar should lead to deprecated in model. maybe again a thing that should be opt-out-able

Reply:
Ok. But I think this is a nice feature to do language evolution.

Reply:
yes, with an opt out option i can live with that.